Gems/GradientSignal/Code/Source/Components/ImageGradientComponent.cpp

```diff
@@ -238,10 +386,53 @@ namespace GradientSignal
             return;
         }
 
+        // Prevent loading of the image data if an invalid advanced configuration
+        // was specified by the user
+        if (m_configuration.m_advancedMode)
+        {
+            auto numComponents = AZ::RHI::GetFormatComponentCount(format);
+            AZ::u8 channel = aznumeric_cast<AZ::u8>(m_configuration.m_channelToUse);
+            if (m_configuration.m_channelToUse == ChannelToUse::Terrarium)
+            {
+                if (!DoesFormatSupportTerrarium(format))
+                {
+                    AZ_Error("GradientSignal", false, "Unable to interpret image as Terrarium because image asset (%s) has pixel format (%s), which only supports %d channels",
+                        m_configuration.m_imageAsset.GetHint().c_str(), AZ::RHI::ToString(format), numComponents);
+                    return;
+                }
+            }
+            else if (channel >= numComponents)
+            {
+                auto channelOptions = SupportedChannelOptions();
+                AZ_Error("GradientSignal", false, "Unable to use channel (%s) because image asset (%s) has pixel format (%s), which only supports %d channels",
+                    channelOptions[channel].m_description.c_str(), m_configuration.m_imageAsset.GetHint().c_str(), AZ::RHI::ToString(format), numComponents);
+                return;
+            }
+
+            // Validate manual scale
+            if (m_configuration.m_customScaleType == CustomScaleType::Manual)
+            {
+                m_configuration.m_scaleRangeMin = AZStd::clamp(m_configuration.m_scaleRangeMin,
+                    aznumeric_cast<float>(AZStd::numeric_limits<AZ::u8>::lowest()),
+                    aznumeric_cast<float>(AZStd::numeric_limits<AZ::u8>::max()));
+                m_configuration.m_scaleRangeMax = AZStd::clamp(m_configuration.m_scaleRangeMax,
+                    aznumeric_cast<float>(AZStd::numeric_limits<AZ::u8>::lowest()),
+                    aznumeric_cast<float>(AZStd::numeric_limits<AZ::u8>::max()));
+                m_configuration.m_scaleRangeMax = AZStd::max(m_configuration.m_scaleRangeMax, m_configuration.m_scaleRangeMin);
```

Comment:
Is it safe to assume the sub image is always and 8 bit type here?

Reply:
Nope I made a bad assumption here :) Will fix up this logic (similar comment below from mbalfour-amzn)